x-pack/plugins/cases/public/components/cases_context/use_application.tsx

```diff
@@ -5,18 +5,27 @@
  * 2.0.
  */
 import useObservable from 'react-use/lib/useObservable';
+import type { BehaviorSubject, Observable } from 'rxjs';
 import { useKibana } from '../../common/lib/kibana';
 
 interface UseApplicationReturn {
   appId: string | undefined;
   appTitle: string | undefined;
 }
 
+/**
+ * Checks if the observable is stateful and, in case, sets up `useObservable` with an initial value
+ */
+const useStatefulObservable = <T,>(observable: Observable<T>) => {
+  const defaultValue = 'value' in observable ? (observable as BehaviorSubject<T>).value : undefined;
```

Comment:
What do you think if we do

```
const isBehaviorSubjectObservable = <T,>(observable: Observable<T>): observable is BehaviorSubject<T> => 'value' in observable

const useStatefulObservable = <T,>(observable: Observable<T>) => {
 if (isBehaviorSubjectObservable(observable)) {
    return useObservable(observable, observable.value);
  }
  
  return useObservable(observable);
};
```

Reply:
I like the type guard! I'd have to use a variable for the initial value thought, because we cannot call hooks conditionally